pkg/cloud/talos/provisioners/packet/packet.go

```diff
@@ -0,0 +1,227 @@
+		udStruct.Networking.OS.Devices = append(udStruct.Networking.OS.Devices, map[string]interface{}{"interface": "eth0", "dhcp": true})
+	}
+
+	udBytes, err := yaml.Marshal(udStruct)
+	if err != nil {
+		return err
+	}
+	//TODO(rsmitty): Shebang no longer needed once talos alpha 28 is cut.
+	ud := "#!talos\n" + string(udBytes)
+
+	devCreateReq := &packngo.DeviceCreateRequest{
+		Hostname:      machine.ObjectMeta.Name,
+		Plan:          packetConfig.Instances.Plan,
+		Facility:      []string{packetConfig.Instances.Facility},
+		OS:            "custom_ipxe",
+		BillingCycle:  "hourly",
+		ProjectID:     packetConfig.ProjectID,
+		UserData:      ud,
+		IPXEScriptURL: packetConfig.Instances.PXEUrl,
+	}
+
+	dev, _, err := packet.client.Devices.Create(devCreateReq)
+	if err != nil {
+		return err
+	}
+
+	//Wait for masters to be active, attach floating ip
+	if isMaster {
+		err = packet.waitForStatus(machine, "active")
+		if err != nil {
+			return err
+		}
+		ipReq := &packngo.AddressStruct{Address: floatingIP + "/32"}
+		_, _, err = packet.client.DeviceIPs.Assign(dev.ID, ipReq)
+		if err != nil {
+			return err
+		}
+	}
+
+	log.Println("[Packet] Instance created with id: " + dev.ID)
+
+	return nil
+}
+
+//Update updates a given Packet instance.
+func (packet *Packet) Update(cluster *clusterv1.Cluster, machine *clusterv1.Machine, clientset *kubernetes.Clientset) error {
+	return nil
+}
+
+// Delete deletes a Packet instance.
+func (packet *Packet) Delete(cluster *clusterv1.Cluster, machine *clusterv1.Machine, clientset *kubernetes.Clientset) error {
+
+	dev, err := packet.fetchDevice(machine)
+	if err != nil {
+		return err
+	}
+
+	if dev != nil {
+		_, err = packet.client.Devices.Delete(dev.ID)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Exists returns whether or not an instance is present in AWS.
+func (packet *Packet) Exists(cluster *clusterv1.Cluster, machine *clusterv1.Machine, clientset *kubernetes.Clientset) (bool, error) {
+
+	dev, err := packet.fetchDevice(machine)
+	if err != nil {
+		return false, err
+	}
+
+	if dev == nil {
+		return false, nil
+	}
+	return true, nil
+}
+
+func (packet *Packet) fetchDevice(machine *clusterv1.Machine) (*packngo.Device, error) {
+	machineSpec, err := utils.MachineProviderFromSpec(machine.Spec.ProviderSpec)
+	if err != nil {
+		return nil, err
+	}
+	packetConfig := &ClusterInfo{}
+	yaml.Unmarshal([]byte(machineSpec.Platform.Config), packetConfig)
+
+	devList, _, err := packet.client.Devices.List(packetConfig.ProjectID, &packngo.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, dev := range devList {
+		if dev.Hostname == machine.ObjectMeta.Name {
+			return &dev, nil
+		}
+	}
+	return nil, nil
+}
+
+//waitForStatus polls the Packet api for a certain instance status
+//needed for attaching elastic IP after boot
+func (packet *Packet) waitForStatus(machine *clusterv1.Machine, desiredState string) error {
+
+	timeout := time.After(300 * time.Second)
```

Comment:
I think it's worth increasing this timeout a bit more, probably at least 600s to be safe. Usually things are pretty quick, but I've run into instances where you need to wait a few moments/minutes for devices to be returned to the pool before provisioning can begin.

Reply:
Fixed